### DISCUSSION TITLE
How to customize picked image data  as a base 64 string to upload to WebAPIs

### DISCUSSION THREAD

**Discussion by PhyoeKo** [singerdmx/flutter-quill#718] [General]
I am using this package as a text editor, there user can write both text and image as well, then I produce result to Html code. There I am getting html img attribute data is picked file path. Likes  `<img src="/data/user/0/com.example.app/app_flutter/image_picker6905649150262524545.jpg" alt="" />`

I want to replace base64 string instead of file path, is there any way to achieve my issues.
Thanks you.

**Comment by AlissonMMenezes** [comment]
Just found a way, you can do it customize the ImageButton like this

```
QuillToolbarImageButton(
                      controller: _controller,
                      options: QuillToolbarImageButtonOptions(
                          imageButtonConfigurations:
                              QuillToolbarImageConfigurations(
                        onRequestPickImage:
                            (context, imagePickerService) async {
                          var imgFile = await imagePickerService.pickImage(
                              source: imgSource.ImageSource.gallery);
                          var bytes = await imgFile!.readAsBytes();

                          // Encode the file data to base64
                          String base64String = base64Encode(bytes);
                          return base64String;
                        },
                        onImageInsertCallback: (image, controller) async {

                          controller.insertImageBlock(
                              imageSource: "data:image/png;base64,${image}");
                        },
                      )),
                    ),
```

**Reply by nguyenlong1999** [reply]
thanks

**Reply by AiMAnsarie** [reply]
Hey Alisson,

Thanks for your response but it looks like in the version 11.4.0, the 2nd parameter (imagePickerService) of onRequestPickImage method has been removed. I haven't been able to find a way to control the image upload as I have just started using Quill packages. Would you be able to provide a solution in the light of this change?

Thanks!
